### DISCUSSION TITLE
fitView doesn't seem to work when the tab containing ReactFlow is hidden

### DISCUSSION THREAD

**Discussion by kobe651jp** [xyflow/xyflow#2373] [General]
fitView doesn't seem to work when the tab containing ReactFlow is hidden. Also defaultZoom does not work.

My guess is that the DOM hasn't finished rendering, so the size can't be measured and zooming might not work properly.

So I tried using an IntersectionObserver to call instance.fitView when the element is visible. It works, but you can see that the movement is unnatural, even though it looks different.

Is this behavior normal? Am I missing something?

https://user-images.githubusercontent.com/44263972/185576466-7a500f6d-ad69-482c-b518-f0e7592c5660.mp4

**Comment by moklick** [comment]
I would try to only render the ReactFlow container when the tab is visible. That should work and also gives your app a better performance. There is no need to render the container when it's not visible.

**Reply by kobe651jp** [reply]
Thank you for answering.
